node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -105,12 +115,40 @@ func (atp *apiTransactionProcessor) doGetTransaction(hash []byte, withResults bo
 	return atp.getTransactionFromStorage(hash)
 }
 
-func (atp *apiTransactionProcessor) populateProcessingTypeFields(tx *transaction.ApiTransactionResult) {
+func (atp *apiTransactionProcessor) populateComputedFieldsProcessingType(tx *transaction.ApiTransactionResult) {
 	typeOnSource, typeOnDestination := atp.txTypeHandler.ComputeTransactionType(tx.Tx)
 	tx.ProcessingTypeOnSource = typeOnSource.String()
 	tx.ProcessingTypeOnDestination = typeOnDestination.String()
 }
 
+func (atp *apiTransactionProcessor) populateComputedFieldInitiallyPaidFee(tx *transaction.ApiTransactionResult) {
+	initiallyPaidFee, err := atp.feeComputer.ComputeTransactionFee(tx.Tx, int(tx.Epoch))
+	if err != nil {
+		log.Warn("populateInitiallyPaidFee(): unexpected condition, cannot compute fee", "tx", tx.Hash)
+		return
+	}
+
+	isZero := initiallyPaidFee.Cmp(big.NewInt(0)) == 0
+	if isZero {
+		return
```

Comment:
why not setting tx.InitiallyPaidFee to "0" and then return? Will be more consistent when consuming the request.

Reply:
Previously, I wanted to not have the field in the response if that's zero ("0" will not be omitted by the serializer). However, on a second thought, changed to:

```
func (atp *apiTransactionProcessor) populateComputedFieldInitiallyPaidFee(tx *transaction.ApiTransactionResult) {
	// Only user-initiated transactions will present an initially paid fee.
	if tx.Type == string(transaction.TxTypeNormal) || tx.Type == string(transaction.TxTypeInvalid) {
		fee := atp.feeComputer.ComputeTransactionFee(tx.Tx, int(tx.Epoch))
		tx.InitiallyPaidFee = fee.String()
	}
}
```

@iulianpascalau, @bogdan-rosianu, @miiu96, is this all right?

Reply:
actually, for 0 it will still return "" because it's a `big.Int` :man_facepalming: